File: pinyougou_web_user/src/main/java/cn/itcast/core/controller/UserController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.user.User;
import cn.itcast.core.service.UserService;
import com.alibaba.dubbo.config.annotation.Reference;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import util.PhoneFormatCheckUtils;

@RestController
@RequestMapping("/user")
public class UserController {

    @Reference
    private UserService userservice;

    /**
     * 发送验证码
     * @param phone
     * @return
     */
    @RequestMapping("/sendCode")
    public Result sendCode(String phone){
        //判断手机号是否合法
        if(PhoneFormatCheckUtils.isPhoneLegal(phone)){
            try {
                userservice.sendCode(phone);
                return new Result(true,"发送成功");
            } catch (Exception e) {
                e.printStackTrace();
                return new Result(false,"发送失败");
            }
        }else {
            return new Result(false,"手机号不合法");
        }
    }

    /**
     * 用户注册
     * @param user
     * @param smscode
     * @return
     */
    @RequestMapping("/add")
    public Result add(@RequestBody User user,String smscode){
        try {
            userservice.add(user,smscode);
            return new Result(true,"注册成功");
        } catch (RuntimeException e) {
            return new Result(false,e.getMessage());
        } catch (Exception e){
            e.printStackTrace();
            return new Result(false,"注册失败");
        }
    }
}

File: pinyougou_web_user/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller   ,userService){	
	$scope.entity = {};
	//注册用户
	$scope.reg=function(){
		//账号密码不能为空
		if($scope.entity.username == null || $scope.entity.password == null || $scope.password == null){
			alert("用户名密码不问为空");
			return;
		}

		//比较两次输入的密码是否一致
		if($scope.password!=$scope.entity.password){
			alert("两次输入密码不一致，请重新输入");
			$scope.entity.password="";
			$scope.password="";
			return ;			
		}
		//新增
		userService.add($scope.entity,$scope.smscode).success(
			function(response){
				alert(response.message);
			}		
		);
	}
    
	//发送验证码
	$scope.sendCode=function(){
		if($scope.entity.phone==null || $scope.entity.phone==""){
			alert("请填写手机号码");
			return ;
		}
		
		userService.sendCode($scope.entity.phone  ).success(
			function(response){
				alert(response.message);
			}
		);		
	}
	
});	
